src/legacy/ui/public/field_editor/components/field_format_editor/editors/duration/duration.tsx

```diff
@@ -18,30 +18,63 @@
  */
 
 import React, { Fragment } from 'react';
+import { DurationFormat } from 'src/plugins/data/common';
 
 import { EuiFieldNumber, EuiFormRow, EuiSelect } from '@elastic/eui';
 
-import { DefaultFormatEditor } from '../default';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+import {
+  DefaultFormatEditor,
+  defaultState,
+  FormatEditorProps,
+  FormatEditorState,
+} from '../default';
 
 import { FormatEditorSamples } from '../../samples';
 
-import { FormattedMessage } from '@kbn/i18n/react';
-import { i18n } from '@kbn/i18n';
+interface DurationFormatEditorState {
+  hasDecimalError: boolean;
+}
 
-export class DurationFormatEditor extends DefaultFormatEditor {
-  static formatId = 'duration';
+interface InputFormat {
+  kind: string;
+  text: string;
+}
 
-  constructor(props) {
-    super(props);
-    this.state.sampleInputs = [-123, 1, 12, 123, 658, 1988, 3857, 123292, 923528271];
-    this.state.hasDecimalError = false;
-  }
+interface OutputFormat {
+  method: string;
+  text: string;
+}
+
+interface DurationFormatEditorFormatParams {
+  outputPrecision: number;
+  inputFormat: string;
+  outputFormat: string;
+}
+
+export class DurationFormatEditor extends DefaultFormatEditor<
+  DurationFormatEditorFormatParams,
+  DurationFormatEditorState
+> {
+  static formatId = 'duration';
+  state = {
+    ...defaultState,
```

Comment:
I'm thinking about this from the perspective of "is there the potential for unwanted shared state?" Since instances of the classes will be created I think the answer is no. Yes, its possible that code might be shared in an unwanted manner somewhere down the line but 🤷